packages/drivers/odsp-driver/src/test/odspDriverUrlResolver2Tests.spec.ts

```diff
@@ -0,0 +1,142 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable max-len */
+/* eslint-disable dot-notation */
+
+import { strict as assert } from "assert";
+import sinon from "sinon";
+import { IRequest } from "@fluidframework/core-interfaces";
+import { OdspDriverUrlResolver2 } from "../odspDriverUrlResolver2";
+import { SharingLinkScopeFor, SharingLinkTokenFetcher } from "../tokenFetch";
+import { getHashedDocumentId } from "../odspUtils";
+import { createOdspUrl } from "../createOdspUrl";
+import * as graph from "../graph";
+import { getLocatorFromOdspUrl, storeLocatorInOdspUrl } from "../odspFluidFileLink";
+import { IOdspResolvedUrl, SharingLinkHeader } from "../contracts";
+
+describe("OdspUrlResolver3", () => {
+    const siteUrl = "https://microsoft.sharepoint-df.com/siteUrl";
+    const driveId = "driveId";
+    const itemId = "fileId";
+    const dataStorePath = "dataStorePath";
+    const fileName = "fileName";
+    const sharelink = "https://microsoft.sharepoint-df.com/site/SHARELINK";
+    let urlResolver: OdspDriverUrlResolver2;
+
+    beforeEach(() => {
+        const shareLinkTokenFetcher: SharingLinkTokenFetcher =
+            async (siteURL: string, scopeFor: SharingLinkScopeFor, refresh: boolean) => "SharingLinkToken";
+            urlResolver = new OdspDriverUrlResolver2(shareLinkTokenFetcher);
+    });
+
+    async function mockFetch<T>(response: Promise<string>, callback: () => Promise<T>): Promise<T> {
```

Comment:
you can take response: string and wrap it with promise here - call callers have resolved string

Reply:
Actually I wanted the callers to have control to put response as a rejected promise. So I am just using a promise here instead of always returning a resolved promise.